### PR TITLE
config: enable the format checker for tidb repos

### PR DESCRIPTION
**Pull request by Mini256**
## Summary
The format checker will make sure the PR body contains the issue number line, the rules will be enabled for the following repos:

- pingcap/tidb
- pingcap/tidb-tools
- pingcap/tidb-binlog
- pingcap/tiflow
- pingcap/ng-monitoring

## Details

1. If there is no Issue Number line in PR Body, the robot will make a prompt and add `do-not-merge/needs-linked-issue` label to block PR from merging.

image

2. In the PR body, we can use the specified keyword to reference the relevant issue. Please refer to the document for details of syntax：[referring-to-an-issue](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue). (Regex test on: https://regex101.com/r/XyJhrT/1)

image

3. When the PR author fills out the issue number line, the robot will remove the `do-not-merge/needs-link-issue` label and prompt comments.

image

4. After the PR is merged, the robot will automatically extract the Issue Number in the PR Body fill in the message body of the squashed commit.

image

### CODE REVIEW

**Comment by Mini256** [comment]
cc: @zhangyangyu

File: prow/config/external_plugins_config.yaml [hunk 1]
```diff
@@ -1365,3 +1365,28 @@ ti-community-format-checker:
...
+  - repos:
+      - pingcap/tidb
+      - pingcap/tidb-tools
+      - pingcap/tidb-binlog
+      - pingcap/ticdc
+      - pingcap/ng-monitoring
+    required_match_rules:
+      - pull_request: true
+        body: true
+        regexp: "(?i)Issue Number:\\s*((,\\s*)?(ref|close[sd]?|resolve[sd]?|fix(e[sd])?)\\s*#(?P<issue_number>[1-9]\\d*))+\\n"
```
Comment (Mini256) [inline prow/config/external_plugins_config.yaml:1377]:
```suggestion
        regexp: "(?i)Issue Number:\\s*((,\\s*)?(ref|close[sd]?|resolve[sd]?|fix(e[sd])?)\\s*#(?P<issue_number>[1-9]\\d*))+(\\r|\\n)"
```

**Review by crazycs520 — COMMENTED** [review]
LGTM

**Comment by Mini256** [comment]
@amyangfei @kennytm @tiancaiamao @lance6716 PTAL~

Please also make sure that the PR modified PR template in each repository to be merged, avoiding the rules of the bot reminder and the PR template does not match.

- https://github.com/pingcap/tidb/pull/30817
- https://github.com/pingcap/tidb-tools/pull/568
- https://github.com/pingcap/tidb-binlog/pull/1099
- https://github.com/pingcap/tiflow/pull/3968
- https://github.com/pingcap/ng-monitoring/pull/65

**Comment by lance6716** [comment]
> config: ignore the PR on release branches or created before 2021/12/27

cc @amyangfei @glorv

**Review by lance6716 — APPROVED** [review]

**Pull request merged**